### PR TITLE
New Feature: User defined showing of tooltips

### PR DESCRIPTION
**Pull request by elliotgordonrowe**
Adds the option for users to define whether to display tooltips over node/metabolite labels or objects (line segments or node circles). enable_tooltips now accepts an array containing either 'object', 'label', both, or neither. Includes the necessary changes to BuilderSettingsMenu.js and API documentation. Also fixes a previously unnoticed regression where touching the canvas wouldn't hide the active tooltip.

### CODE REVIEW

**Comment by zakandrewking** [comment]
We will need to document the change in API here:

https://github.com/spaghettikinase/escher/blob/7c3934b85bc3ca3918acba0e599a3b4f20a46bc4/docs/javascript_api.rst

You can do that as part of the PR.

**Comment by zakandrewking** [comment]
docs look great!

I will still review the code. just haven't gotten to it today

File: src/Draw.js [hunk 1]
```diff
@@ -418,6 +424,23 @@ function update_segment (update_selection, scale, cobra_model,
         return null
       }
     })
+    .attr('pointer-events', 'visibleStroke')
+    .on('mouseover', function (d) {
+      // Add the current mouse position to the segment's datum
+      object_mouseover_fn('reaction_object', Object.assign(
+        {}, d, {xPos: d3_mouse(this)[0], yPos: d3_mouse(this)[1]}
```
Comment (zakandrewking) [inline src/Draw.js]: It's more efficient to call d3_mouse once and assign the result to a variable

File: src/Draw.js [hunk 2]
```diff
@@ -418,6 +424,23 @@ function update_segment (update_selection, scale, cobra_model,
...
+      ))
+    })
+    .on('touchend', function (d) {
+      // Add last touch position to the segment's datum
+      object_touch_fn('reaction_object', Object.assign(
+        {}, d, {xPos: d3_touch(this.parentNode, 0)[0], yPos: d3_touch(this.parentNode, 0)[1]}
```
Comment (zakandrewking) [inline src/Draw.js]: Same stuff d3_touch here

**Comment by zakandrewking** [comment]
And i just tried it out. works perfectly :)

What do you think the best default is, `['label']` or `['label', 'object']`?

**Comment by elliotgordonrowe** [comment]
I think maybe that setting enable_tooltips to both labels and objects might make things a little cluttered but users could always turn off one or the other.

**Pull request merged**